brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java

```diff
@@ -106,8 +108,15 @@ public void supportsPortableCustomization() throws Exception {
             public <T> void response(HttpAdapter<?, T> adapter, T res, Throwable error,
                                      SpanCustomizer customizer) {
                 super.response(adapter, res, error, customizer);
+                // TODO: is there a way to get the URL visible earlier? Waiting until response is
```

Comment:
I probably missed the review comment, but it seems in the current implementation, the URL isn't available until response parsing. this will subvert the intent of most parsers and such. Since we are post-dating the span's timestamp anyway, I wonder if we can delay until the URL is available, but not so late as when the request is already sent. then, we can parse at that moment...

wdyt? @kojilin

Reply:
ex if this were organized differently, or if internals were shifted such that the URL is available earlier, the code simplifies to `adapter.url(request)`

Reply:
Not sure if this is what you're looking for, but how about adding a listener to `RequestLog` so that you are called back when request headers are ready? There's no need to wait until the response.

```java
ctx.log().addListener(log -> {
    generateUrl(log);
}, RequestLogAvailability.REQUEST_HEADERS);
```

Reply:
thanks will try

Reply:
https://github.com/line/armeria/issues/2089